include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -4349,101 +4349,70 @@ namespace internal
   template <int dim, typename Number>
   struct CellwiseInverseMassMatrixImplTransformFromQPoints
   {
-    template <int fe_degree, int = 0>
+    template <int fe_degree, int n_q_points_1d>
     static bool
     run(const unsigned int                  n_desired_components,
         const FEEvaluationBaseData<dim,
                                    typename Number::value_type,
                                    false,
                                    Number> &fe_eval,
         const Number *                      in_array,
-        Number *                            out_array,
-        typename std::enable_if<fe_degree != -1>::type * = nullptr)
+        Number *                            out_array)
     {
-      const AlignedVector<Number> &inverse_shape =
-        fe_eval.get_shape_info().data.front().inverse_shape_values_eo;
+      static const bool do_inplace =
+        fe_degree > -1 && (fe_degree + 1 == n_q_points_1d);
 
-      constexpr unsigned int dofs_per_cell = Utilities::pow(fe_degree + 1, dim);
-      internal::EvaluatorTensorProduct<internal::evaluate_evenodd,
+      Assert(fe_eval.get_shape_info().element_type !=
+               MatrixFreeFunctions::ElementType::tensor_none,
+             ExcNotImplemented());
```

Comment:
For now I have added an assert here, since the code is not working simplices. In a follow-up PR, I intend to add an alternative code path!